### PR TITLE
Create GitHub Endpoints

### LINKED ISSUE #16: Create endpoints for getting information from GitHub
Some information that should be fetched are:
* commits
* pull requests
* issues

### PR DESCRIPTION
**Pull request by jackcmeyer**
closes #16 

* adds functionality to get commits, pull requests, and issues from a github repository

merge #32 before

### CODE REVIEW

File: test/projects.service.test.js [hunk 1]
```diff
@@ -7,11 +7,14 @@ import { getMockLogger } from './mockLogger';
...
 import { populateUsers, populateProjects } from '../src/db/import_test_data';
+import GithubAuthenticationService from '../src/services/githubauth.service';
+import { getActiveLogger } from '../src/utils/winston';
 
 sequelize.initSequelize();
 
 const userService = new UserService(sequelize.User, sequelize.Credentials, getMockLogger());
-const projectService = new ProjectService(sequelize.Project, userService, getMockLogger());
+const githubAuthService = new GithubAuthenticationService(sequelize.GithubToken, userService, getActiveLogger());
+const projectService = new ProjectService(sequelize.Project, userService, githubAuthService, getMockLogger());
```
Comment (clarkerican) [inline test/projects.service.test.js]: Do we have plans to later add tests for these?

Comment (jackcmeyer) [reply test/projects.service.test.js]: Yeah, if I do them now, i won't get anything done 👎

File: src/controllers/projects.controller.js [hunk 2]
```diff
@@ -157,6 +157,77 @@ export async function deleteProjectById(req, res, next) {
...
+ */
+export async function getPullRequestsForProject(req, res, next) {
+  const projectId = req.params.projectId;
+  const state = req.query.state;
+  const limit = req.query.limit;
+  const user = req.user.id;
+
+  try {
+    const pullRequests = await projectService.getPullRequestsForProject(projectId, state, user);
+    if (limit) {
```
[the remark this answers is not in the document]
Comment (jackcmeyer) [reply src/controllers/projects.controller.js]: We can add this if needed, but I think `limit` will be good for our use case.

File: src/routes/projects.routes.js [hunk 3]
```diff
@@ -374,6 +374,127 @@ router.delete(
...
+     "title": "the title goes here",
+     "description": "the description goes here",
+     "created_at_date": "2018-02-09T16:10:58Z",
+     "closed_at_date": null
+  }
+ ]
+ */
+router.get(
+  '/projects/:projectId/pullrequests',
+  authController.isAuthenticated,
```
[the remark this answers is not in the document]
Comment (jackcmeyer) [reply src/routes/projects.routes.js]: no, should just be the same as viewing any other information

**Pull request merged**